jcl/src/jdk.management/share/classes/com/ibm/lang/management/internal/ExtendedThreadMXBeanImpl.java

```diff
@@ -67,4 +67,95 @@ public ExtendedThreadInfo[] dumpAllExtendedThreads(boolean lockedMonitors, boole
 		return resultArray;
 	}
 
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public long getThreadAllocatedBytes(long threadId) {
+		throw new UnsupportedOperationException();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public long[] getThreadAllocatedBytes(long[] threadIds) {
+		throw new UnsupportedOperationException();
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public long[] getThreadCpuTime(long[] threadIds) {
+		long[] result = new long[threadIds.length];
+		for (int i = 0; i < threadIds.length; i++) {
+			if (threadIds[i] <= 0) {
+				/*[MSG "K05F7", "Thread id must be greater than 0."]*/
+				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K05F7")); //$NON-NLS-1$
+			}
+			result[i] = -1;
+		}
+		if (isCurrentThreadCpuTimeSupported()) {
+			if (isThreadCpuTimeEnabled()) {
+				for (int i = 0; i < threadIds.length; i++) {
+					result[i] = getThreadCpuTime(threadIds[i]);
+				}
+			}
+		} else {
+			/*[MSG "K05F6", "CPU time measurement is not supported on this virtual machine."]*/
+			throw new UnsupportedOperationException(com.ibm.oti.util.Msg.getString("K05F6")); //$NON-NLS-1$
+		}
+		return result;
+	}
+
+	/**
+	 * {@inheritDoc}
+	 */
+	@Override
+	public long[] getThreadUserTime(long[] threadIds) {
+		long[] result = new long[threadIds.length];
+		for (int i = 0; i < threadIds.length; i++) {
+			if (threadIds[i] <= 0) {
+				/*[MSG "K05F7", "Thread id must be greater than 0."]*/
+				throw new IllegalArgumentException(com.ibm.oti.util.Msg.getString("K05F7")); //$NON-NLS-1$
+			}
+			result[i] = -1;
+		}
```

Comment:
This is not consistent with the documented behavior: it is supposed to be equivalent to calling `getThreadUserTime(long)` for each thread id, which would through `NullPointerException` or `UnsupportedOperationException` if `!isCurrentThreadCpuTimeSupported()`.

Reply:
I'm not sure I understand. Are you referring to the IllegalArgumentException? The spec says `IllegalArgumentException - if id <= 0.` for getThreadUserTime(long) and it's the same for long[], `IllegalArgumentException - if any element in the input array ids is <= 0.`.

Reply:
Nm, I see your point.

Reply:
No actually I don't, pls clarify.

Reply:
The documentation suggests the behavior should be as if the implementation were:
```java
		int idCount = threadIds.length;
		long[] result = new long[idCount];
		for (int i = 0; i < idCount; i++) {
			result[i] = getThreadUserTime(threadIds[i]);
		}
		return result;
```
* if `threadIds == null` then it will throw `NullPointerException`
* if `threadIds[i] <= 0` then it will throw `IllegalArgumentException`
* if `!isCurrentThreadCpuTimeSupported()` then it will throw `UnsupportedOperationException`

A difference in the proposed implementation is that all thread ids are considered before any check of `isCurrentThreadCpuTimeSupported()`. This seems consistent with the [documented reasons](https://github.com/ibmruntimes/openj9-openjdk-jdk11/blob/openj9/src/jdk.management/share/classes/com/sun/management/ThreadMXBean.java#L62-L63) that `IllegalArgumentException` might be thrown. I retract my objection.